deeplake/core/vectorstore/utils.py

```diff
@@ -0,0 +1,76 @@
+import deeplake
+from deeplake.constants import MB
+
+import numpy as np
+
+from typing import Dict
+
+
+def create_or_load_dataset(dataset_path, token, creds, logger, read_only, **kwargs):
+    if dataset_exists(dataset_path, token, creds, **kwargs):
+        return load_dataset(dataset_path, token, creds, logger, read_only, **kwargs)
+    return create_deeplake_dataset(dataset_path, token, **kwargs)
+
+
+def dataset_exists(dataset_path, token, creds, **kwargs):
+    return (
+        deeplake.exists(dataset_path, token=token, **creds)
+        and "overwrite" not in kwargs
+    )
+
+
+def load_dataset(dataset_path, token, creds, logger, read_only, **kwargs):
+    dataset = deeplake.load(dataset_path, token=token, read_only=read_only, **kwargs)
+
+    logger.warning(
+        f"Deep Lake Dataset in {dataset_path} already exists, "
+        f"loading from the storage"
+    )
+    dataset.summary()
+    return dataset
+
+
+def create_deeplake_dataset(dataset_path, token, **kwargs):
```

Comment:
I think we should not cement us into this fixed structure of tensors and let users be able to dynamically later add arbitrary datasets.